substrate/state-machine/src/ext.rs

```diff
@@ -58,17 +59,29 @@ pub struct Ext<'a, B: 'a> {
 	pub backend: &'a B,
 }
 
+#[cfg(test)]
+impl<'a, B: 'a + Backend> Ext<'a, B> {
+	pub fn storage_pairs(&self) -> Vec<(Vec<u8>, Vec<u8>)> {
+		self.backend.pairs().iter()
+			.map(|&(ref k, ref v)| (k.to_vec(), Some(v.to_vec())))
+			.chain(self.overlay.committed.clone().into_iter())
+			.chain(self.overlay.prospective.clone().into_iter())
+			.collect::<HashMap<_, _>>()
+			.into_iter()
```

Comment:
the collect and reiteration seem redundant

Reply:
ordering doesn't matter, removing uniques in a last-come-stays does.